### PR TITLE
fix #67291: Confirmation before deletion of a palette

### PR DESCRIPTION
**Pull request by salilkhandelwal**
There will be a confirmation message before deleting a palette in custom workspaces.

### CODE REVIEW

File: mscore/palettebox.cpp [hunk 1]
```diff
@@ -191,18 +191,23 @@ void PaletteBox::paletteCmd(PaletteCommand cmd, int slot)
...
-                  delete item;
-                  item = vbox->itemAt(slot);
-                  vbox->removeItem(item);
-                  delete item->widget();
-                  delete item;
-                  for (int i = 0; i < (vbox->count() - 1) / 2; ++i)
-                        static_cast<PaletteBoxButton*>(vbox->itemAt(i * 2)->widget())->setId(i*2);
-                  emit changed();
+                  QMessageBox::StandardButton reply;
+                  reply = QMessageBox::question(this," Warning "," Are you sure ? ",QMessageBox::Yes | QMessageBox::No);
```
Comment (Jojo-Schmitz) [inline mscore/palettebox.cpp]: use `... QWidget::tr("Warning"), QWidget::tr"Are you sure?"` ... to (a) make the textx translatable, (b) remove the superfluos spaces

File: mscore/palettebox.cpp [hunk 2]
```diff
@@ -191,18 +191,23 @@ void PaletteBox::paletteCmd(PaletteCommand cmd, int slot)
...
+                  if (reply==QMessageBox::Yes) {
```
Comment (Jojo-Schmitz) [inline mscore/palettebox.cpp]: spaces around == please

**Comment by Jojo-Schmitz** [comment]
No reason to close PR, just change and ``git push -f`

**Comment by Jojo-Schmitz** [comment]
Ignore thre Travis failure for now, it doesn't seem to be related to your work at all

**Comment by salilkhandelwal** [comment]
Okay :)

File: mscore/palettebox.cpp [hunk 3]
```diff
@@ -191,18 +191,23 @@ void PaletteBox::paletteCmd(PaletteCommand cmd, int slot)
...
-                  delete item;
-                  item = vbox->itemAt(slot);
-                  vbox->removeItem(item);
-                  delete item->widget();
-                  delete item;
-                  for (int i = 0; i < (vbox->count() - 1) / 2; ++i)
-                        static_cast<PaletteBoxButton*>(vbox->itemAt(i * 2)->widget())->setId(i*2);
-                  emit changed();
+                  QMessageBox::StandardButton reply;
+                  reply = QMessageBox::question(this,QWidget::tr("Warning"),QWidget::tr("Are you sure?"),QMessageBox::Yes | QMessageBox::No);
```
Comment (salilkhandelwal) [inline mscore/palettebox.cpp]: @Jojo-Schmitz only 1 space after each comma ?

Comment (Jojo-Schmitz) [inline mscore/palettebox.cpp]: yes, see also https://musescore.org/en/developers-handbook/musescore-coding-rules#Binary-Operators

**Pull request merged**